dask/array/utils.py

```diff
@@ -65,13 +65,6 @@ def meta_from_array(x, ndim=None, dtype=None):
     if isinstance(x, type):
         x = x(shape=(0,) * (ndim or 0), dtype=dtype)
 
-    if (
-        not hasattr(x, "shape")
-        or not hasattr(x, "dtype")
-        or not isinstance(x.shape, tuple)
-    ):
-        return x
-
     if isinstance(x, list) or isinstance(x, tuple):
```

Comment:
I moved this before the `not hasattr(x, "shape") or not hasattr(x, "dtype")`, because otherwise it's a no-op: lists and tuples don't have `shape`/`dtype`, so we always would have returned from the function before reaching this check.